Sources/SotoCore/Message/AWSRequest.swift

```diff
@@ -169,7 +170,8 @@ extension AWSRequest {
                     path = path
                         .replacingOccurrences(of: "{\(location)}", with: Self.urlEncodePathComponent(String(describing: value)))
                         .replacingOccurrences(of: "{\(location)+}", with: Self.urlEncodePath(String(describing: value)))
-
+                    hostPrefix = hostPrefix?
+                        .replacingOccurrences(of: "{\(location)}", with: Self.urlEncodePathComponent(String(describing: value)))
```

Comment:
I would prefer to add a new location `.host` which affects the hostname but that would be a breaking change. So I'm using `.uri` which kinda makes sense

Reply:
Yeah I think it's about as good as it can be